src/ab-test-preview.js

```diff
@@ -2,58 +2,79 @@ import { getLetter } from '../src/utils';
+		const totalTemplates = templates.length;
 
-		// set the button inner html to the test title
-		tab.innerHTML = templates[i].dataset.title;
+		for ( let i = 0; i < totalTemplates; i++ ) {
+			// create a button element
+			const tab = document.createElement( 'button' );
 
-		// if the title is empty the set the title based on the index
-		if ( templates[i].dataset.title === '' ) {
-			tab.innerHTML = sprintf( __( 'Variant %s', 'altis-analytics' ), getLetter( i ) );
-		}
+			// if the title is empty the set the title based on the index
+			if ( templates[i].dataset.title === '' ) {
+				tab.innerHTML = sprintf( __( 'Variant %s', 'altis-analytics' ), getLetter( i ) );
+			} else {
+				// set the button inner html to the test title
+				tab.innerHTML = templates[i].dataset.title;
+			}
+
+			// set the class for the button
+			tab.className = 'ab-test-xb-preview__tab';
+
+			tab.addEventListener( 'click', function ( e ) {
+				// clone the correspnding data for the tab clicked
+				const variant = templates[i].content.cloneNode( true );
 
-		// set the class for the button
-		tab.className = 'ab-test-xb-preview__tab';
+				// remove the active class from any tab
+				tabContainer.querySelectorAll( '.ab-test-xb-preview__tab' ).forEach( el => {
+					el.classList.remove( 'active' );
+				} );
 
-		tab.addEventListener( 'click', function ( e ) {
-			// clone the correspnding data for the tab clicked
-			const variant = templates[i].content.cloneNode( true );
+				// add the active class
+				tab.classList.add( 'active' );
+				tabContent.innerHTML = '';
 
-			// remove the active class from any tab
-			tabContainer.querySelectorAll( '.ab-test-xb-preview__tab' ).forEach( el => {
-				el.classList.remove( 'active' );
+				// append the data
+				tabContent.appendChild( variant );
 			} );
 
-			// add the active class
-			tab.classList.add( 'active' );
-			tabContent.innerHTML = '';
+			// append the created tab
+			tabContainer.appendChild( tab );
+		}
 
-			// append the data
-			tabContent.appendChild( variant );
-		} );
+		// determine if a specific tab should be clicked
+		// else click the first tab
+		if ( urlTest && ( urlTest[2] < totalTemplates ) ) {
+			tabContainer.children[urlTest[2]].click();
+		} else {
+			tabContainer.firstElementChild.click();
+		}
 
-		// append the created tab
-		tabContainer.appendChild( tab );
-	}
+	} );
+}
 
-	// determine if a specific tab should be clicked
-	// else click the first tab
-	if ( url_test ) {
-		tabContainer.children[url_test[2]].click();
-	} else {
-		tabContainer.firstElementChild.click();
-	}
+// Create the tabs for previewing
+createTabbedPreviews();
 
+// When making changes in the customizer
+wp.customize.selectiveRefresh.bind( 'sidebar-updated', function () {
+	// clear out any exsisting tabs
+	document.querySelectorAll( '.ab-test-xb-preview' ).forEach( xb => {
+		const tabContainer = xb.querySelector( '.ab-test-xb-preview__tabs' );
+		tabContainer.innerHTML = '';
+	} );
```

Comment:
When making a chance in the customizer it would add the tabs to the exsisting tabs, so essentially duplicate them and give me something like this:
![Screenshot 2022-01-03 at 15 25 31](https://user-images.githubusercontent.com/16571365/147935695-8ca9dc1c-74ab-4617-a435-d5a0c21108b0.png)

Reply:
Right but you could check in the function if they're already there rather than resetting.